src/NzbDrone.Core/DecisionEngine/Specifications/RssSync/HistorySpecification.cs

```diff
@@ -116,6 +124,41 @@ public virtual DownloadSpecDecision IsSatisfiedBy(RemoteEpisode subject, SearchC
                             return DownloadSpecDecision.Reject(DownloadRejectionReason.HistoryUpgradesNotAllowed, "{0} grab event in history and Quality Profile '{1}' does not allow upgrades", rejectionSubject, qualityProfile.Name);
                     }
                 }
+
+                if (episode is { HasFile: true })
+                {
+                    EpisodeHistory availableUsableEpisodeHistoryForCustomFormatScore;
+
+                    var episodeHistoryEventTypeForHistoryComparison = new List<EpisodeHistoryEventType>
+                    {
+                        EpisodeHistoryEventType.Grabbed,
+                        EpisodeHistoryEventType.SeriesFolderImported,
+                    };
+
+                    if (episodeHistoryEventTypeForHistoryComparison.Contains(mostRecent.EventType))
+                    {
+                        availableUsableEpisodeHistoryForCustomFormatScore = mostRecent;
```

Comment:
You can use a ternary for this:

```
var availableUsableEpisodeHistoryForCustomFormatScore = episodeHistoryEventTypeForHistoryComparison.Contains(mostRecent.EventType) ? mostRecent : _historyService.MostRecentForEpisodeInEventCollection(
                            episode.Id,
                            new ReadOnlyCollection<EpisodeHistoryEventType>(episodeHistoryEventTypeForHistoryComparison));
```

I'd say that these variables are too verbose in naming though, they're monstrosities to read.